src/core_plugins/input_control_vis/public/control/control.js

```diff
@@ -133,6 +133,14 @@ export class Control {
     return this._kbnFilter;
   }
 
+  setUnsetValue(unsetValue) {
+    const hasValue = this.hasValue();
+    this.filterManager.setUnsetValue(unsetValue);
```

Comment:
This ordering looks important, since changing it around would affect the logic.  Can you add some comments?  It's not very easy to follow.

What should happen, for instance, if the previous `unsetValue` is 1, and the previous `value` is 1, but the new `unsetValue` is 2?  In this logic, `value` is changed to 2.  But should it be?  Because based on the new settings, 1 is a valid value, not an "unset value", so maybe it should stay at 1 and not be reset to 2?

Almost seems like `unsetValue` should be readonly and set at the time of construction because it doesn't really make any sense to allow it to be changed later on (or you can get into that weird situation above).

Reply:
`InputRange` does not support an empty state (where no range is selected). Therefore, we had to come up with some way to handle the empty state. We did this by setting the value to an `unsetValue`. We chose an unset state of setting the min and max to the range slider min. It is a value the user would never set but is still valid via the rules of `InputRange`.

In your example, `hasValue` would be false because the logic in `hasValue()` compares the value to the `unsetValue` (before updating the unsetValue), and returns false when the value equals the unsetValue. The key is that the hasValue is calculated before changing the `unsetValue`. That way, after updating the `unsetValue`, we know that the value is in the empty state and needs to be updated to the new empty state.

Does that make sense?

Reply:
> Almost seems like unsetValue should be readonly and set at the time of construction because it doesn't really make any sense to allow it to be changed later on (or you can get into that weird situation above)

It can not be readonly because it will change as the range min and max changes - otherwise you get these warnings because the value is outside of the bounds of the min and max

Reply:
@Stacey-Gammon I added some tests which should clear up how everything works